### PR TITLE
Fix Attribute#getJavaType() returs also primitive types

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/PrimitiveBasicTypeImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+package org.hibernate.metamodel.model.domain.internal;
+
+import org.hibernate.type.descriptor.java.JavaType;
+
+public class PrimitiveBasicTypeImpl<J> extends BasicTypeImpl<J> {
+	private final Class<J> javaTypeClass;
+
+	public PrimitiveBasicTypeImpl(JavaType<J> javaType, Class<J> javaTypeClass) {
+		super( javaType );
+		this.javaTypeClass = javaTypeClass;
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/PrimitiveBasicTypeImpl.java:16]: Minor, but would be nice to add an assertion here that `javaTypeClass.isPrimitive()`

**Review by sebersole — COMMENTED** [review]
LGTM

File: hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/PrimitiveBasicTypeImpl.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/PrimitiveBasicTypeImpl.java:17]: Done!

**Comment by karge-itestra** [comment]
@dreab8 Is there an issue associated to this? I am seeing this issue in 5.2 as well, will it be backported? It appeared to me in a curious place: when extracting the values in a tuple, the implementation makes an instanceof check for the given element type and the requested type (the requested type comes from a metamodel attribute), which fails for example for `Long` as actual type and `long` as requested type.

**Comment by beikov** [comment]
There is no issue for this @karge-itestra. If you face a problem in 5.x, please create a JIRA issue with a test case against 5.6, as 5.4 and 5.5 are not maintained anymore.

**Pull request merged**